src/sagas/backend.ts

```diff
@@ -201,6 +201,32 @@ function* backendSaga(): SagaIterator {
     }
   });
 
+  /**
+   * Unsubmits the submission and updates the grading overviews of the new status.
+   */
+  yield takeEvery(actionTypes.UNSUBMIT_SUBMISSION, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+    const { submissionId } = (action as actionTypes.IAction).payload;
+    const overviews = yield select((state: IState) => state.session.gradingOverviews || []);
+    const resp: Response = yield postUnsubmit(submissionId, tokens);
```

Comment:
Again use a guard clause here. It is more efficient. If not, you are unnecessary looping through the `overviews` when your resp has returned an error.

```
if (!resp || !resp.ok) {
     //handle error
}

//main logic
```

Reply:
Done